File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string, vector<string>> clothes_map;
    
    for (int i = 0; i < clothes.size(); i ++){
        clothes_map[clothes[i][1]].push_back(clothes[i][0]);
    }
    for (auto it: clothes_map){
        answer  *= it.second.size() + 1;
    }
    
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String[] cloth: clothes) {
            map.put(cloth[1], map.getOrDefault(cloth[1], 0) + 1);
        }
        
        int answer = 1;
        for (String key: map.keySet()) {
            answer *= map.get(key) + 1;
        }
        
        return answer - 1;
    }
}